### PR TITLE
Add production and testing code for Env to master branch

### PR DESCRIPTION
**Pull request by Soul-ze**

### CODE REVIEW

**Review by yehanz — APPROVED** [review]
Great job in general, Mingze!

Only a few concerns that I think worth being added to our team's TODO list:
Considering training process could take tens of thousands of rounds of play, the performance issue of our Qttt might be team's next focus, where we need to refactor some of our heavy-duty functions to make it as efficient as possible. But this is a good start point and not our top priority until we complete our deep RL part.

In summary, thank you for your hard work!
